fuel-vm/src/interpreter/ecal.rs

```diff
@@ -88,4 +88,38 @@ where
             Ok(())
         }
     }
+
+    /// Read access to the ECAL state
+    pub fn ecal_state(&self) -> &Ecal {
+        &self.ecal_state
+    }
+
+    /// Write access to the ECAL state
+    pub fn ecal_state_mut(&mut self) -> &mut Ecal {
+        &mut self.ecal_state
+    }
+
+    /// Replace the ECAL state and handler.
+    /// Useful when the handler does not implement `Default`.
+    pub fn with_ecal<NewEcal: EcalHandler>(
```

Comment:
It would be nice to have a constructor that allows you also to create an instance of the `Interpreter` with `Ecal` that doesn't support `Default`.

Because right now you need to create `Interpreter` with empty `Ecal` and replace it after, which is strange=)

Reply:
If we have a new constructor, do we really need that method then?

Reply:
Changed to work like this in 6f32f3f0.